woocommerce/woocommerce-gutenberg-products-block
# Move all totals components into components package

## What

Moves the `Subtotal`, `TotalsFees`, and `TotalsItem` components into the `@woocommerce/blocks-components` package.

It also fixes some TS errors in some files.
 
Fixes #11687

## Why

It doesn't make sense to have all of the components spread across three directories.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Install [Extra Fees Plugin for WooCommerce](https://wordpress.org/plugins/woo-conditional-product-fees-for-checkout/)
2. Set it up like so and save your changes: (Please choose a product in your store if you don't have "Belt")
<img width="915" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5656702/16b3d2b5-1bec-4769-95c5-b28d53c44ce0">

3. Set up taxes in your store and ensure some items are taxed.
4. Add these items, and the item with a fee from step 2 to your cart.
5. Go to the Cart block and see your "Extra fee" line.
6. Ensure all the other totals elements appear, e.g. Taxes, Subtotal, Shipping, and Total.
7. Go to the Checkout block and check the same elements appear correctly.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:

* [ ] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor. **No UI Change, but tested in Safari and Firefox anyway**

## Changelog


> Move `TotalsItem`, `TotalsFees`, and `Subtotal` components to the `@woocommerce/blocks-components` package.
